enhancements/microshift/dummy-gateway-interface-ovn.md

```diff
@@ -0,0 +1,476 @@
+
+#### Configuring
+
+There is no need to introduce any new configuration flag for this change. Unless
+we want to allow user to specify the IP address used by the br-ex interface.
+
+#### Deploying Applications
+
+Applications are deployed as usual.
+
+#### Variation [optional]
+
+N/A
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+On the OVN-Kubernetes side, we need to change the local gateway mode:
+
+1. Allow using an external gateway bridge without uplink port
+2. Add static routes for all host network with the ovn-k internal masquerade
+   next-hop.
+
+On the MicroShift side, we will need to make the following change:
+
+1. We need to update the microshift-ovs-init.service to 
+   * Create br-ex bridge, and assign a static IP to it.
+   * Since we don't need NetworkManager to manage the DHCP client. We can remove
+   those logic from the configure-ovs.sh.
+   * For upgrading, we also need to remove the NetworkManager connections
+   created by configure-ovs.sh in previous version.
+
+### Risks and Mitigations
+
+The OVN-Kubernetes configuration described in this document is not used by any
+other known platforms. Therefore there is no CI with this setup in
+OVN-Kubernetes upstream. In order to minimize the risk of regression, we need to
+improve the OVN-Kubernetes CI to cover this setup.
+
+### Drawbacks
+
+The egress features that rely on sending packets to wire without passing kernel
+will not work. As there is no uplink port to forward those packets to. So
+egressGW and egressIP will not work. A possible solution is to forward those
+traffic to the host kernel, then adding iptables rules to steer the traffic to
+wire.
+
+With this change, the ovn node annotations such
+`k8s.ovn.org/node-primary-ifaddr` and `k8s.ovn.org/l3-gateway-config` will use
+the local IP of br-ex as the node IP, instead of the IP of host's primary
+interface. Therefore the following features that need to read these node
+annotation will be affected. The egress rules of Egress firewall contains the
+node IP. And the traffic case of services with `internalTrafficPolicy=Local`
+backed by hostNetwork pods will break. For services with
+`internalTrafficPolicy=Local`, ovn-k will only put local host endpoints to the
+node switch loadbalancer. If the pod is hostNetwork, the local endpoint means
+the pod with the NodeIP. Normally, the ovn-k will use the nodeIP as the ovn-k
+external IP in node annotation `k8s.ovn.org/l3-gateway-config`. However, in our
+case, as we use a dummy interface for br-ex, we have different IPs for nodeIP
+and ovn-k external IP. So ovn-k cannot identify the right local endpoint in such
+a case. It then put no endpoint to the load-balancer.This problem is not so
+critical to MicroShift. Since we only get one node in the cluster. Creating a
+service with `internalTrafficPolicy=Local` makes no sense in such a case. We can
+fix this problem later until we find a proper solution.
```

Comment:
Since we are adding explicit support in OVN-K for no uplink, should we find friendly ways to report to the user the features that are not working. 
For example, update egress IP status with a relevant message, etc...

Reply:
Good point.